src/Build/Logging/SerialConsoleLogger.cs

```diff
@@ -471,6 +537,17 @@ public override void ErrorHandler(object sender, BuildErrorEventArgs e)
             SetErrorsOrWarningsOnCurrentFrame();
             ShowDeferredMessages();
             setColor(ConsoleColor.Red);
+
+            //determine the mapping of properties to output
+            Dictionary<string, string> outputProperties = new Dictionary<string, string>();
+            if (e.BuildEventContext != null)
+            {
+                int nodeId = e.BuildEventContext.NodeId;
+                int projectContextId = e.BuildEventContext.ProjectContextId;
+                TargetFramework_mapping.TryGetValue((nodeId, projectContextId), out outputProperties);
```

Comment:
```suggestion
                TargetFramework_mapping.TryGetValue((nodeId, projectContextId), out e.outputProperties);
```

If you set it here like this, you can set it below with:
`e.outputProperties ??= new Dictionary<string, string>();`
and avoid allocating when e.BuildEventContext != null. Also, when is e.BuildEventContext null? Similar examples above.

Reply:
I hadn't realized that BuildEventContext could be null and then when I had tried running the tests some of the tests had it null which I assume is a rare case but still possible. TryGetValue oddly enough can't take a property or indexer. Otherwise this would be a great idea!